### PR TITLE
Backport when condition in validators from yii2

### PR DESCRIPTION
**Pull request by miramir**

### CODE REVIEW

**Comment by samdark** [comment]
@miramir looks OK. Could you work on docs for it?

File: framework/validators/CValidator.php [hunk 1]
```diff
@@ -109,6 +109,25 @@
...
+	 * This property is mainly provided to support conditional validation on the server side.
+	 * If this property is not set, this validator will be always applied on the server side.
+	 *
+	 * The following example will enable the validator only when the country currently selected is USA:
+	 *
+	 * ```php
+	 * function ($model) {
+	 *     return $model->country == Country::USA;
+	 * }
+	 * ```
```
Comment (samdark) [inline framework/validators/CValidator.php]: Need `@since 1.1.17`.

File: framework/validators/CValidator.php [hunk 2]
```diff
@@ -198,7 +217,9 @@ public function validate($object,$attributes=null)
 		foreach($attributes as $attribute)
 		{
 			if(!$this->skipOnError || !$object->hasErrors($attribute))
-				$this->validateAttribute($object,$attribute);
+				if ($this->when === null || call_user_func_array($this->when, array($object, $attribute))) {
```
Comment (samdark) [inline framework/validators/CValidator.php]: Formatting is not consistent with 1.1.

File: framework/validators/CValidator.php (hunk 1 above)
Comment (miramir) [inline framework/validators/CValidator.php]: fixed

**Comment by miramir** [comment]
@samdark к сожалению я плохо говорю по английский. И нормально написать документацию не смогу.

**Comment by samdark** [comment]
OK, I'll write it then.

**Comment by samdark** [comment]
Closing since pull request is messed up and we aren't accepting features anymore.

**Pull request closed without merging**